src/BlockTypes/ProductAddToCartButton.php

```diff
@@ -0,0 +1,262 @@
+	 *
+	 * @param string $block_content HTML content of the block being rendered.
+	 * @param array  $block         The full block, including name and attributes.
+	 */
+	public function on_render_block( $block_content, $block ) {
+		if ( 'core/button' !== $block['blockName'] ) {
+			return $block_content;
+		}
+
+		global $post;
+		$post_id = $post->ID;
+
+		if ( $this->is_woocommerce_variation( $block ) ) {
+			$product = wc_get_product( $post_id );
+
+			if ( $product ) {
+				$styles_and_classes = $this->extract_style_and_class_from_block_content( $block_content );
+				$div_class          = $styles_and_classes['div_class'];
+				$div_style          = $styles_and_classes['div_style'];
+				$anchor_class       = $styles_and_classes['anchor_class'];
+				$anchor_style       = $styles_and_classes['anchor_style'];
+
+				return apply_filters(
+					'woocommerce_loop_add_to_cart_link',
+					'<div class="wp-block-button wc-block-components-product-button wc-block-grid__product-add-to-cart ' . $div_class . '" style="' . $div_style . '">
+							<a
+								href="' . esc_url( $product->add_to_cart_url() ) . '"
+								rel="nofollow"
+								data-product_id="' . esc_attr( $product->get_id() ) . '"
+								data-product_sku="' . esc_attr( $product->get_sku() ) . '"
+								class="wp-block-button__link ' . ( $product->is_purchasable() && ! $product->has_options() ? 'ajax_add_to_cart add_to_cart_button' : '' ) . ' product_type_' . esc_attr( $product->get_type() ) . ' ' . $anchor_class . '"
+								style="' . esc_attr( $anchor_style ) . '">'
+									. '<span>'
+									. $this->get_anchor_inner_html( $block_content, $product )
+									. '</span>'
+							. '</a>'
+					. '</div>',
+					$product
+				);
+			}
+		}
+
+		return $block_content;
+	}
+
+	/**
+	 * Extracts style and class from block content
+	 *
+	 * @param string $block_content The HTML content of the block.
+	 *
+	 * @return array
+	 */
+	private function extract_style_and_class_from_block_content( $block_content ) {
+		if ( empty( $block_content ) ) {
+			return array(
+				'div_class'    => '',
+				'div_style'    => '',
+				'anchor_class' => '',
+				'anchor_style' => '',
+			);
+		}
+
+		$dom = new DOMDocument();
+		$dom->loadHTML( $block_content );
+
+		$div       = $dom->getElementsByTagName( 'div' )->item( 0 );
+		$div_class = $div->getattribute( 'class' );
+		$div_style = $div->getattribute( 'style' );
+
+		$div          = $dom->getElementsByTagName( 'a' )->item( 0 );
+		$anchor_class = $div->getattribute( 'class' );
+		$anchor_style = $div->getattribute( 'style' );
+
+		return array(
+			'div_class'    => $div_class,
+			'div_style'    => $div_style,
+			'anchor_class' => $anchor_class,
+			'anchor_style' => $anchor_style,
+		);
+	}
+
+
+	/**
+	 * Return inner html of anchor block. Some examples of inner html of `a` tag:
+	 * - <strong><em><mark><code>Add to cart</code></mark></em></strong>
+	 * - <strong>Add</strong> <strong><em>to</em></strong> <em>cart</em>
```

Comment:
```suggestion
	 * Return inner html of anchor block.
	 * 
	 * Due to the rich text capabilities of the Button block, the markup can look
	 * quite convoluted. Here are some examples of the inner HTML of `a` tag:
	 * ```html
	 * - <strong><em><mark><code>Add to cart</code></mark></em></strong>
	 * - <strong>Add</strong> <strong><em>to</em></strong> <em>cart</em>
	 *```
```

Reply:
Thanks for making this comment more clear 🙌🏻